### PR TITLE
[DO NOT MERGE] Library search refactor in preparation for public library search

### PR DESCRIPTION
**Pull request by yipingliao**
@ahsu1230 @andrewconner 
https://app.asana.com/0/18079737520750/18807532114522

This pull refactors the current library search to use a different url. Instead of `https://www.kifi.com/find?q=car&f=a&l=l9Rre2MSpAPb`, we now have `https://www.kifi.com/lydialaurenson/halloween/find?q=car&f=a`.

Please do not merge yet; I still need to work on the permissions part.

@ahsu1230 Your review is much appreciated, since I changed a lot of code in `HeaderCtrl`.

Thanks!

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/search/search.js [hunk 1]
```diff
@@ -67,6 +41,46 @@ angular.module('kifi')
...
+        $scope.hasMore = true;
+        $scope.scrollDistance = '100%';
+        $scope.loading = false;
+
+        $window.document.title = 'Kifi • ' + query;
+
+        searchActionService.reset();
+        $scope.getNextKeeps(true);
+      })['catch'](function (resp) {
+        if (resp.status && resp.status === 403) {
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/search/search.js:81]: What should we do when a user tries an url that searches a library that the user has no access to? We could redirect to login, show the not-found page, or show the need-to-login page (that prompts for the passphrase).

File: kifi-backend/modules/shoebox/angular/src/search/search.js [hunk 2]
```diff
@@ -3,18 +3,22 @@
...
     //
     var query;
-    var filter = $routeParams.f || 'm';
-    var library = $routeParams.l || '';
-
+    var filter;
+    var library;
     var lastResult = null;
     var selectedCount = 0;
+    var authToken = $location.search().authToken || $location.search().authCode || $location.search().accessToken || '';
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/search/search.js]: We should only be using use `authToken` now. The other `authCode` or `accessToken` were when we were still confused / deciding the name lol

File: kifi-backend/modules/shoebox/angular/src/search/search.js (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/search/search.js:81]: Not found would be closest in line to what we do currently.

File: kifi-backend/modules/shoebox/angular/src/search/search.js (hunk 2 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/search/search.js]: Will do. Thanks!

File: kifi-backend/modules/shoebox/angular/src/search/search.js (hunk 1 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/search/search.js:81]: Okay, will do.

**Pull request closed without merging**